src/xrpld/ledger/detail/View.cpp

```diff
@@ -2123,4 +2492,65 @@ rippleCredit(
         saAmount.asset().value());
 }
 
+static Number
+getShareTotal(ReadView const& view, std::shared_ptr<SLE> const& vault)
+{
+    auto issuance =
```

Comment:
Should we check if nullptr?

Reply:
Excellent point. If we check it, what do we do if the object isn't there ? I believe that should be `tefINTERNAL` but in order to return an error, we need to pull the check outside of this function. So I removed this function entirely and reactored the functions using it, by pulling the issuance object read operation all the way to the `doApply`.